packages/autocomplete/src/containers/AutocompleteContainer.js

```diff
@@ -0,0 +1,568 @@
+    /**
+     * The currently focused tag
+     */
+    tagFocusedKey: PropTypes.string,
+    /**
+     * Returns all controlled state. Used for controlling usage.
+     */
+    onStateChange: PropTypes.func,
+    /**
+     * These placements differ from the default naming of Popper.JS placements to help
+     * assist with RTL layouts.
+     */
+    placement: PropTypes.oneOf([
+      'auto',
+      'top',
+      'top-start',
+      'top-end',
+      'end',
+      'end-top',
+      'end-bottom',
+      'bottom',
+      'bottom-start',
+      'bottom-end',
+      'start',
+      'start-top',
+      'start-bottom'
+    ]),
+    /**
+     * Passes options to [Popper.JS Instance](https://github.com/FezVrasta/popper.js/blob/master/docs/_includes/popper-documentation.md#new-popperreference-popper-options)
+     */
+    popperModifiers: PropTypes.object,
+    /**
+     * Identical to children
+     */
+    render: PropTypes.func,
+    /**
+     * @param {Object} renderProps
+     * @param {Function} renderProps.getInputProps - Props to be spread onto the input element
+     * @param {Function} renderProps.getTriggerProps - Props to be spread onto the containing, trigger element
+     * @param {Function} renderProps.getTagProps - Props to be spread onto each tag element. `({ key })` is required
+     * @param {Function} renderProps.triggerRef - Callback for the ref of the element triggering the menu
+     * @param {Function} renderProps.inputRef - Callback for the ref of the input
+     * @param {Boolean} renderProps.isOpen - Whether the Menu is currently visible
+     * @param {Boolean} renderProps.tagFocusedKey - The key of the currently focused tag
+     */
+    trigger: PropTypes.func,
+    /**
+     * The z-index of the popper.js placement container
+     */
+    zIndex: PropTypes.number
+  };
+
+  static defaultProps = {
+    placement: 'bottom-start',
+    eventsEnabled: true,
+    zIndex: 1000
+  };
+
+  constructor(...args) {
+    super(...args);
+
+    this.state = {
+      isOpen: false,
+      id: IdManager.generateId(),
+      focusedKey: undefined,
+      tagFocusedKey: undefined
+    };
+
+    this.focusSelectionModel = new SingleSelectionModel();
+    this.focusSelectionModel.onSelectionChanged = this.onFocusSelectionModelChange;
+
+    this.tagSelectionModel = new SingleSelectionModel();
+    this.tagSelectionModel.onSelectionChanged = this.onTagSelectionModelChange;
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { focusedKey } = this.getControlledState();
+    const previousFocusedKey = this.isControlledProp('focusedKey')
+      ? prevProps.focusedKey
+      : prevState.focusedKey;
+
+    /**
+     * We must programatically scroll the newly focused element into view.
+     * Side-effect of the `aria-activedescendant` accessibility strategy.
+     */
+    if (typeof focusedKey !== 'undefined' && focusedKey !== previousFocusedKey) {
+      const itemNode = document.getElementById(this.getItemId(focusedKey));
```

Comment:
If we push through the [document context branch](https://github.com/zendeskgarden/react-components/compare/zendeskgarden:e192dc6...zendeskgarden:04610b8) I have we'll need to update this too

Reply:
I agree. Whichever solution we go with we will have to make sure that it will work with all of the components, not just select/menu.

Are you ok with handling this in a follow-on task since it seems out-of-scope for this?

Reply:
For posterity, #49 has been merged so this can use the `getDocument` util now.

Reply:
Rebased, updated all the dependencies, and added in the `getDocument` utility